lib/manager/deps-edn/extract.ts

```diff
@@ -16,7 +16,7 @@ export function extractPackageFile(content: string): PackageFile {
     match = regex.exec(rest);
 
     deps.push({
-      datasource: datasourceClojure.id,
+      datasource: new ClojureDatasource().id,
```

Comment:
we should export the id also directly, so we do not need those changes 😕 Looks poretty bad to instanciate for id property only

Reply:
I _think_ I have a better solution for this. I define `Datasource` as

```ts
export abstract class Datasource implements DatasourceApi {
  protected constructor(id: string) {
    this.http = new Http(id);
    this.id = id;
  }
}
```

and implement it like so:

```ts
export class CdnJsDatasource extends Datasource {
  static readonly id = 'cdnjs';

  constructor() {
    super(CdnJsDatasource.id);
  }
}
```

The main issue is this doesn't actually require `id` to be defined. I could put an assert in the constructor that would be caught in unit tests, but throwing exceptions in constructors is generally considered bad practice.

I can't think of a good way to get the compiler to enforce this constraint.

Reply:
Would need strict null check in typescript. But that would need a lot of work. 😉

Reply:
Simpler:

```ts
export abstract class Datasource implements DatasourceApi {
  protected http: Http;
  protected constructor(protected readonly id: string) {
    this.http = new Http(id);
  }
}
```